app/services/assessment_particulars_creation_service.rb

```diff
@@ -0,0 +1,20 @@
+class AssessmentParticularsCreationService
+  def initialize(remote_ip, request_payload)
+    @request_payload = request_payload
+    @remote_ip = remote_ip
+  end
+
+  def call
+    assessment = Assessment.create!(remote_ip: @remote_ip, request_payload: @request_payload)
+    hash = {
+      request: JSON.parse(@request_payload),
+      response: {
+        assessment_id: assessment.id,
+        client_reference_id: assessment.client_reference_id,
+        details: {},
+        errors: []
+      }
+    }
+    RecursiveOpenStruct.new(hash, recurse_over_arrays: true)
+  end
```

Comment:
Define getter methods so that you don't have to access instance variables directly:
```ruby
attr_reader :remote_ip, :request_payload
```

Then break down `call` so it isn't doing too much:
```ruby
def call
  RecursiveOpenStruct.new(data, recurse_over_arrays: true)
end

def assessment 
  @assessment ||= Assessment.create!(remote_ip: remote_ip, request_payload: request_payload)
end

def data
  {
    request: JSON.parse(request_payload),
    response: {
      assessment_id: assessment.id,
      client_reference_id: assessment.client_reference_id,
      details: {},
      errors: []
    }
  }
end
```
Then expose the service functionality via a class method - to ensure that items used in the construction aren't shared across calls
```ruby
def self.create_for(*args)
  new(*args).call
end
```
Then to create a new set of Assessment Particulars:
```ruby
AssessmentParticularsCreationService.create_for remote_ip, request_payload
```